chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+
+% Why it is a separate process -------------------------------
+
+% From cleaning
+Construction is done separately from data cleaning for two reasons. 
+The first one is to clear differentiation between the data originally collected and the result of data processing decisions.
+The second is to ensure that variable definition is consistent across data sources. 
+Unlike cleaning, construction can create many outputs from many inputs. 
+Let's take the example of a project that has a baseline and an endline survey. 
+Unless the two instruments are exactly the same, which is unlikely,  the data cleaning for them will require different steps, and therefore will be done separately. 
+However, you still want the constructed variables to be calculated in the same way, so they are comparable.
+So you want to construct indicators for both rounds in the same code, after merging them.
+
+% From analysis
 Data construction is never a finished process.
-It comes ``before'' data analysis only in a limited sense:
-the construction code must be run before the analysis code.
-Typically, however it is the case that the construction and analysis code
-are written concurrently -- both do-files will be open at the same time.
-It is only in the process of writing the analysis
-that you will come to know which constructed variables are necessary,
-and which subsets and alterations of the data are desired.
-
-You should never attempt to create a ``canonical'' analysis dataset.
-Each analysis dataset should be purpose-built to answer an analysis question.
-It is typical to have many purpose-built analysis datasets:
+It comes ``before'' data analysis only in a limited sense: the construction code must be run before the analysis code.
+Typically, however, construction and analysis code are written concurrently.
+As you write the analysis, different constructed variables will become necessary, as well as subsets and other alterations to the data.
+Still, constructing variables in a separate script from the analysis will help you ensure consistency across different outputs. 
+If every script that creates a table starts by loading a data set, subsetting it and manipulating variables, any edits to construction need to be replicated in all scripts. 
+This increases the chances that at least one of them will have a different sample or variable definition.
+
+
+% What to do during construction -----------------------------------------
+Construction is the step where you face the largest risk of making a mistake that will affect your results. 
+Keep in mind that details and scales matter. 
+It is important to check and double-check the value-assignments of questions and their scales before constructing new variables using them. 
+Are they in percentages or proportions? 
+Are all variables you are combining into an index or average using the same scale? 
+Are yes or no questions coded as 0 and 1, or 1 and 2?
```

Comment:
_Are yes or no questions **consistently** coded as 0 and 1, or 1 and 2?_

Reply:
What about "binary" questions rather than yes/no? (Thinking also of gender-type stuff where I've seen very basic mistakes get made)

Reply:
Ok, I see that this is not clear. But not sure how to address it. It doesn't really matter in this example if they are consistently coded, or if they are categorical or binary. It just matters that you're mindful of the underlying value. Will try to rephrase it.

Reply:
```suggestion
In Stata, categorical variables have underlying numerical values, so it is important to be mindful of what they are when using this type of variable: a variable with "Yes" and "No" options, for example, may be coded as 0 and 1, or just as well as 1 and 2.
```

Reply:
While possible to code differently, isn't very error prone? Should we recommend that all yes/no is coded into 1/0 so they numerically can be used as frequencies in means and dummies in regressions?

Reply:
Yes, you're right. We should.

Reply:
Yes we should but we should leave this part as is, so something like:

> Are yes/no questions coded as 0 and 1, or 1 and 2? Make sure they are recoded to 0 and 1. Similarly, you should typically generate dummies for each category in multiple-response variables (such as gender).

Reply:
Moved to issue #345